Decouple next.js webpage structure from components

ThanGerlek opened the pull request:
Extracted essentially everything from all the page.tsx files into their own components, which should make dependency management and component testing easier. The only remaining dependency on next.js as far as I can tell is the navigation logic (useRouter() and the <Link /> component), which would also be useful to extract eventually but isn't vital for now.

ThanGerlek committed:
feat: add AuthenticationService.ts

ThanGerlek committed:
feat: add LoginPresenter.ts and connect to component

ThanGerlek committed:
feat: add SignupPresenter and refactor out commonalities

ThanGerlek committed:
Merge remote-tracking branch 'origin/main' into auth-presenters

ThanGerlek committed:
refactor: mv everything out of /app that's not a next.js webpage

ThanGerlek committed:
refactor: rename page.tsx files (breaking change, but only until the
next commit)

ThanGerlek committed:
fix: add page.tsx files back in (which redirect to the previous ones)

ThanGerlek committed:
clean: rmv unneeded "use client" lines

ThanGerlek committed:
refactor: move previous page components out of /app

ThanGerlek committed:
fix: incorrect component name

ThanGerlek committed:
refactor: push url-param handling up into page.tsx

ThanGerlek committed:
fix: broken urls (they were unintentionally refactored)

ThanGerlek committed:
clean: rmv unused import (remnant from a cherry-pick)

johnstewart00 commented on frontend/src/components/pages/Home.tsx:
```diff
@@ -0,0 +1,33 @@
+        <h2 className="text-5xl font-bold">Contacts</h2>
+        <div className="flex justify-end p-4 ml-12">
+          <Link
+            href="../../app/addContact"
```
I couldn't get this link working when I ran the branch locally, I had to change href to be a more of an absolute path

`href="/addContact"`

johnstewart00 commented on frontend/src/components/pages/Home.tsx:
```diff
@@ -0,0 +1,33 @@
+        </div>
+        <div className="flex justify-end p-4 ml-12">
+          <Link
+            href="../../app/addEvent"
```
Same with this line:

`href="/addEvent"`

johnstewart00 reviewed (commented):
Looks great besides these two small changes.

ThanGerlek committed:
Merge branch 'main' into decouple-next-pages (and fix merge conflicts)

ThanGerlek committed:
lint: make sure useEffect() is always called

Previously, it would not be called if a contact was not found. ESLint tag: react-hooks/rules-of-hooks

ThanGerlek committed:
fix: broken links

ThanGerlek committed:
fix: suppress warning for intentional unnecessary local var

ThanGerlek committed:
Merge branch 'main' into auth-presenters

ThanGerlek committed:
eslint: "use" or remove unused variables in stubbed methods

ThanGerlek committed:
eslint: suppress eslint (project-wide) for no-empty-object-type

ThanGerlek committed:
Merge branch 'decouple-next-pages' into auth-presenters (& fix conflicts)

johnstewart00 committed:
Merge pull request #18 from webecke/auth-presenters

Auth presenters
